File: C/seleccion_centinela.c
// Programa donde se busca un elemento por centinela y se ordena usando el algoritmo de selection sort

#include <stdio.h>

// Funcion para llenar un arreglo

void fill_array(int arr[], int size)
{
    register int i;
    int element;

    for (i = 0; i < size; i++)
    {
        printf("\nIngrese el valor del elemento %d ", i);
        scanf("%d", &element);
        arr[i] = element;
    }
    printf("\n");
}

// Funcion para imprimir un arreglo

void print_array(int arr[], int size)
{
    register int i;

    printf("\nLos elementos del arreglo son: [ ");
    
    for (i = 0; i < size; i++)
    {
        printf("%d ",arr[i]);
    }
    
    printf("]\n");
}

// Funcion de busqueda por centinela

int sentinel_search(int arr[], int size, int v)
{
    register int i;

    for (i = 0, arr[size] = v; arr[i] != v; i++);

    if (i < size)
        return i;
    else
        return -1;
}

// Funcion para ordenar por selection sort

void selection_sort(int arr[], int size)
{
    register int i, j;
    int min;

    for (i = 0; i < size - 1; i++)
    {
        min = i;
        for (j = i + 1; j < size; j++)
        {
            if (arr[j] < arr[min])
                min = j;
        }
        if (min != i)
        {
            int aux;
            aux = arr[i];
            arr[i] = arr[min];
            arr[min] = aux;
        }
    }
}

void main()
{
    int n, pos, value;

    printf("\nIngrese la cantidad de elementos del arreglo: ");
    scanf("%d", &n);

    int array[n];

    // Llamada a funcion para rellenar el arreglo

    fill_array(array, n);

    // Llamada a funcion para imprimir el arreglo

    print_array(array, n);

    // Llamada a funcion para buscar un elemento por centinela

    printf("\nIngrese el elemento a buscar en el arreglo: ");
    scanf("%d", &value);

    pos = sentinel_search(array, n, value);
    printf("\nLa posicion del elemento es: %d", pos);

    // Llamada a funcion para ordenar el arreglo por selection sort

    selection_sort(array, n);
    printf("\nOrdenando por seleccion...\n");
    print_array(array, n);
}

File: Python/seleccion_centinela.py
# Programa para buscar un elemento con centinela y ordenarlo usando el algoritmo de selection sort

# Funcion para llenar una lista

def fill_list(lista):
    length = int(input("\nIngrese la cantidad de elementos de la lista: "))

    for i in range(0, length):
        value = int(input(f"Ingrese el elemento numero {i}: "))
        lista.append(value)

# Funcion para la busqueda secuencial con centinela

def sentinel_search(lista, size, v):
    lista.append(v)
    i = 0

    while (lista[i] != v):
        i += 1

    if i < size:
        return i
    else:
        return -1

# Funcion para el selection sort

def selection_sort(lista, size):

    for i in range(0, size - 1):
        min = i
        for j in range(i + 1, size):
            if lista[j] < lista[min]:
                min = j
        
        if min != i:
            aux = lista[i]
            lista[i] = lista[min]
            lista[min] = aux



# Programa main()

int_list = []

# Llamada a funcion para llenar la lista

fill_list(int_list)
list_size = len(int_list)
print(f"\nLa lista es: {int_list}")

# Llamada a funcion para buscar un elemento con centinela

value = int(input("Ingrese el valor del elemento a buscar: "))
pos = sentinel_search(int_list, list_size, value)
print(f"\nLa posicion del elemento {value} es {pos}")
int_list.pop()

# Llamada a funcion para ordenar el arreglo mediante selection sort

print("\nOrdenando mediante selection sort...\n")
selection_sort(int_list, list_size)
print(f"La lista es: {int_list}")